include/nlohmann/json.hpp

```diff
@@ -5599,14 +5691,23 @@ class basic_json
             // check if iterator pos fits to this JSON value
             if (JSON_HEDLEY_UNLIKELY(pos.m_object != this))
             {
-                JSON_THROW(invalid_iterator::create(202, "iterator does not fit current value"));
+                JSON_THROW(invalid_iterator::create(202, "iterator does not fit current value", diagnostics_t(*this)));
             }
 
             // insert to array and return iterator
+#if JSON_DIAGNOSTICS
+            iterator result = insert_iterator(pos, cnt, val);
+            for (size_type i = 0; i < cnt; ++i)
+            {
+                (result + static_cast<typename iterator::difference_type>(i))->m_parent = this;
+            }
+            return result;
+#else
             return insert_iterator(pos, cnt, val);
+#endif
```

Comment:
For the reasons above:

            iterator result = insert_iterator(pos, cnt, val);
    #if JSON_DIAGNOSTICS
            for (typename iterator::difference_type i = 0; i < static_cast<typename iterator::difference_type>(cnt); ++i)
            {
                set_parent(result + i);
            }
    #endif
            return result;

Reply:
This happens enough that it could be a function:

    void set_parents(iterator result, iterator::difference_type count)

Reply:
Good idea, I took the first proposal. See f8037660d06a0fbb6b1cde944d436032d04564ae.